lib/stub_ui/lib/src/ui/text.dart

```diff
@@ -1011,6 +1011,185 @@ enum BoxWidthStyle {
   max,
 }
 
+/// Data class that holds the metrics for a single line in the [Paragraph].
+class LineMetrics {
+
+  @pragma('vm:entry-point')
+  LineMetrics._(
+    this.startIndex,
+    this.endIndex,
+    this.endExcludingWhitespace,
+    this.endIncludingNewline,
+    this.hardBreak,
+    this.ascent,
+    this.descent,
+    this.unscaledAscent,
+    this.height,
+    this.width,
+    this.left,
+    this.baseline,
+    this.lineNumber,
+    List<RunMetrics> rMetrics,
+    List<int> rMetricsIndexes,
+  ) : runMetrics = Map() {
+    for (int i = 0; i < rMetrics.length; ++i) {
+      runMetrics[rMetricsIndexes[i]] = rMetrics[i];
+    }
+  }
+
+  /// The index in the text buffer the line begins.
+  final int startIndex;
+
+  /// The index in the text buffer the line ends.
+  final int endIndex;
+
+  /// The index in the text buffer the line ends ignoring whitespace.
+  ///
+  /// This value is provided as if any whitespace at the end were truncated.
+  final int endExcludingWhitespace;
+
+  /// The index in the text buffer the line ends including newline characters.
+  ///
+  /// This value is provided as if any trailing newline characters for the
+  /// current line were included. Newline characters in other lines are always
+  /// accounted for.
+  final int endIncludingNewline;
+
+  /// True if this line ends with a explicit line break (eg, '\n') or is the end
+  /// of the paragraph. False otherwise.
+  final bool hardBreak;
+
+  // The final computed ascent and descent for the line. This can be impacted by
+  // the strut, height, scaling, as well as outlying runs that are very tall.
+  //
+  // The top edge is `baseline - ascent` and the bottom edge is `baseline +
+  // descent`. Ascent and descent are provided as positive numbers. Raw numbers
+  // for specific runs of text can be obtained in run_metrics_map. These values
+  // are the cumulative metrics for the entire line.
+
+  /// The rise from the [baseline] as calculated from the font and style for this line.
+  ///
+  /// The [ascent] is provided as a positive value, even though it is typically defined in
+  /// fonts as negative. This is to ensure the signage of operations with these
+  /// metrics directly reflects the intended signage of the value. For example,
+  /// the y-coordinate of the top edge of the line is 'baseline - ascent`.
+  final double ascent;
+
+  /// The drop from the [baseline] as calculated from the font and style for this line.
+  ///
```

Comment:
Should this contain the same information about the positiveness?

Reply:
Descent is usually positive by default. I guess I could mention it, but most devs would expect it to be positive in the first place.